### PR TITLE
Checkout matching openshift-ansible branch

### PR DESCRIPTION
**Pull request by jhadvig**
@stevekuznetsov PTAL

Running tests with mentioned changes:
- PR against release 1.5 - https://ci.openshift.redhat.com/jenkins/view/All/job/jhadvig/5/
- PR against master        - https://ci.openshift.redhat.com/jenkins/view/All/job/jhadvig/6/

Fixes https://github.com/openshift/origin/issues/15186

### CODE REVIEW

File: sjb/config/test_cases/test_branch_origin_extended_conformance_install_update.yml [hunk 1]
```diff
@@ -23,6 +23,10 @@ extensions:
       script: |-
         tito_tmp_dir="tito"
         mkdir -p "${tito_tmp_dir}"
+        if [[ ${ORIGIN_TARGET_BRANCH} == release-1.* ]]
+        then
+          git checkout -b ${ORIGIN_TARGET_BRANCH} origin/${ORIGIN_TARGET_BRANCH}
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_origin_extended_conformance_install_update.yml]: Isn't there a bunch of logic for translating branch names? Where Origin has `release-1.x` or `release-3.x` and OpenShift Ansible has `enterprise-x`?

Comment (jhadvig) [inline sjb/config/test_cases/test_branch_origin_extended_conformance_install_update.yml]: I dont think so, cause when you check the origin repo branches you can see what branches are there, in this case only `release-1.x` and `stage-1.x` ... and those reflect to the o-a branches... `enterprise-1.x` branches are only present in OCP

Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_origin_extended_conformance_install_update.yml]: oh, maybe scott moved them

Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_origin_extended_conformance_install_update.yml]: Definitely used to be `enterprise-x`

**Comment by stevekuznetsov** [comment]
I would rather see if we can fix the `oct sync openshift-ansible` step generated by sjb at the start of the job

**Comment by jhadvig** [comment]
Yeah I kinda understand why you want to fix it in the generated step but this is a pretty specific use-case in which we checkout the branch of o-a repository based on the branch against which the PR is sent.  Not sure if this wouldn't be better in the step itself with a proper comment

@stevekuznetsov thoughts ?

File: sjb/config/test_cases/test_branch_origin_extended_conformance_install_update.yml (hunk 1 above)
Comment (jhadvig) [inline sjb/config/test_cases/test_branch_origin_extended_conformance_install_update.yml]: Yeah I was surprised as well, cause I remember they have been there

**Comment by stevekuznetsov** [comment]
Any reason why we don't always want to check out the matching branch if it exists? Seems like this would be applicable to `stage`, too (but not `rebase-1.7`)

**Comment by jhadvig** [comment]
Closing in favor of https://github.com/openshift/aos-cd-jobs/pull/434

**Pull request closed without merging**